src/firefly/java/edu/caltech/ipac/firefly/core/Util.java

```diff
@@ -0,0 +1,205 @@
+import java.util.function.Consumer;
+import java.util.function.Predicate;
+
+/**
+ * Date: 11/19/24
+ *
+ * @author loi
+ * @version : $
+ */
+public class Util {
+    private static final Logger.LoggerImpl logger = Logger.getLogger();
+
+
+    /**
+     * Serializes a Java object into a Base64-encoded string.
+     * <p>
+     * This method converts the given object into a byte stream,
+     * encodes the byte stream into a Base64 string, and returns the result.
+     * The object must implement {@link java.io.Serializable} for this method to work.
+     * </p>
+     * @param obj the object to serialize; must implement {@link java.io.Serializable}
+     * @return a Base64-encoded string representing the serialized object, or null
+     */
+    public static String serialize(Object obj) {
+        if (obj == null) return null;
+        try {
+            ByteArrayOutputStream bstream = new ByteArrayOutputStream();
+            ObjectOutputStream ostream = new ObjectOutputStream(bstream);
+            ostream.writeObject(obj);
+            ostream.flush();
+            byte[] bytes =  bstream.toByteArray();
+            return Base64.getEncoder().encodeToString(bytes);
+        } catch (Exception e) {
+            logger.warn(e);
+            return null;
+        }
+    }
+
+    /**
+     * Deserializes a Base64-encoded string into a Java object.
+     * <p>
+     * This method decodes the provided Base64 string into a byte stream,
+     * then reconstructs the original object using Java's object serialization mechanism.
+     * </p>
+     * @param base64 the Base64-encoded string representing the serialized object
+     * @return the deserialized Java object, or null.
+     */
+    public static Object deserialize(String base64) {
+        try {
+            if (base64 == null) return null;
+            byte[] bytes = Base64.getDecoder().decode(base64);
+            ByteArrayInputStream bstream = new ByteArrayInputStream(bytes);
+            ObjectInputStream ostream = new ObjectInputStream(bstream);
+            return ostream.readObject();
+        } catch (Exception e) {
+            logger.warn(e);
+            return null;
+        }
+    }
+
+
+//====================================================================
+//  Functional Helpers
+//====================================================================
+
+    /**
+     * A function that throws exception
+     * @param <T>  the return type of this function
+     */
+    @FunctionalInterface
+    public interface FuncWithEx<T> {
+        T get() throws Exception;
+    }
+
+    /**
+     * A callable that throws exception
+     */
+    @FunctionalInterface
+    public interface CallWithEx {
+        void run() throws Exception;
+    }
+
+    /**
+     * A function with parameter that throws exception
+     */
+    @FunctionalInterface
+    public interface FuncParamWithEx<P, T> {
+        T apply(P p) throws Exception;
+    }
+
+    public static class Try<T> {
```

Comment:
Can you document these functions, for example, what does Try.it() do? I think it runs something but then returns an empty `Try`. It is not clear why that happens.